### PR TITLE
Bug 2044207: Clear text on switching auth method

### PR DESCRIPTION
**Pull request by weirdwiz**
Clear the value of auth in the state when switching authentication
methods.
Fixes https://bugzilla.redhat.com/show_bug.cgi?id=2044207

### CODE REVIEW

**Comment by afreen23** [comment]
/ok-to-test
cc @GowthamShanmugam

**Comment by weirdwiz** [comment]
cc @SanjalKatiyar

File: frontend/packages/ceph-storage-plugin/src/components/kms-config/vault-config.tsx [hunk 1]
```diff
@@ -64,6 +64,7 @@ export const ValutConfigure: React.FC<KMSConfigureProps> = ({
   };
 
   const setAuthMethod = (authMethod: VaultAuthMethods) => {
+    vaultStateClone.authValue.value = '';
```
Comment (GowthamShanmugam) [inline frontend/packages/ceph-storage-plugin/src/components/kms-config/vault-config.tsx:67]: I dont think the user will switch to a different auth method once he enters details. That is why auth method dropdown is on top, they will decide which auth method to use then they will fill out the form.

I dont think this change is necessary.

Comment (SanjalKatiyar) [reply frontend/packages/ceph-storage-plugin/src/components/kms-config/vault-config.tsx:67]: I think ask for the BZ was to reset entire state when auth method is changed not just resetting the authValue.

Comment (SanjalKatiyar) [reply frontend/packages/ceph-storage-plugin/src/components/kms-config/vault-config.tsx:67]: @GowthamShanmugam do you want to close this BZ ?

Comment (GowthamShanmugam) [reply frontend/packages/ceph-storage-plugin/src/components/kms-config/vault-config.tsx:67]: I dont find any issue with the old value displayed when auth method switches, most of the fields are common and it reduces user effort.  i dont think its a bug.

Comment (SanjalKatiyar) [reply frontend/packages/ceph-storage-plugin/src/components/kms-config/vault-config.tsx:67]: okay then let's talk about authValue... it is not common between two. Shouldn't we at least reset that (will reduce the scope of the BZ from resetting entire state to just value of role/token) ?

Comment (GowthamShanmugam) [reply frontend/packages/ceph-storage-plugin/src/components/kms-config/vault-config.tsx:67]: mm ya it can be reset, i agree

Comment (SanjalKatiyar) [reply frontend/packages/ceph-storage-plugin/src/components/kms-config/vault-config.tsx:67]: okay then this PR looks good... I will comment on the BZ about this as well...

Comment (GowthamShanmugam) [reply frontend/packages/ceph-storage-plugin/src/components/kms-config/vault-config.tsx:67]: ack

**Comment by SanjalKatiyar** [comment]
LGTM

File: frontend/packages/ceph-storage-plugin/src/components/kms-config/vault-config.tsx (hunk 1 above)
Comment (GowthamShanmugam) [reply frontend/packages/ceph-storage-plugin/src/components/kms-config/vault-config.tsx:67]: `setAuthValue('')`

Comment (GowthamShanmugam) [inline frontend/packages/ceph-storage-plugin/src/components/kms-config/vault-config.tsx:67]: > setAuthValue('')

Comment (GowthamShanmugam) [reply frontend/packages/ceph-storage-plugin/src/components/kms-config/vault-config.tsx:67]: Just intializing  vaultStateClone.authValue.value = ''; wont work, value needs to be validated tusing vaultStateClone.authValue.valid = authValue !== '';
setAuthValue('') is alreading doing this.

**Review by GowthamShanmugam — APPROVED** [review]

**Review by bipuladh — APPROVED** [review]

**Pull request merged**